web/pdf_page_view.js

```diff
@@ -260,6 +260,10 @@ class PDFPageView {
    */
   updatePosition() {
     if (this.textLayer) {
+      let readableStream = this.pdfPage.streamTextContent({
+        normalizeWhitespace: true,
+      });
+      this.textLayer.setTextContentStream(readableStream);
```

Comment:
> We just re-requesting data here since we are not storing entire text content anymore. getReader() can be requested once. We can cache entire textContent content after the read, however I want to try to get away without caching.

Sure, but considering that [`PDFPageView.updatePosition`](https://github.com/mozilla/pdf.js/blob/82f3145a5d8630378f378ef8f4f8596204273a6b/web/pdf_page_view.js#L261) is called from [`PDFViewer._scrollUpdate`](https://github.com/mozilla/pdf.js/blob/82f3145a5d8630378f378ef8f4f8596204273a6b/web/pdf_viewer.js#L455) which is called on [*every* scroll event](https://github.com/mozilla/pdf.js/blob/82f3145a5d8630378f378ef8f4f8596204273a6b/web/pdf_viewer.js#L129), couldn't all the (potentially thousands) of `streamTextContent` calls slow us down considerably here?

I'm even thinking that it may explain https://github.com/mozilla/pdf.js/pull/8488#issuecomment-308158072: 
> But unfortunately, it is bit slow to load pages while scrolling and memory usage is also high (as compare to master) :(

Couldn't we at least avoid doing this if the textLayer has already been completely rendered?